src/store/ticTac/reducer.ts

```diff
@@ -25,9 +39,57 @@ export const ticTacReducer: TReducer = (state = initialState, action) => {
         squares: [...state.squares],
       };
     case AT.SET_IS_GAME_END:
+      const setGameEndUpdate: Partial<ITicTac> = {};
+      if (state.myOpponentGame.id) {
+        setGameEndUpdate.myOpponentGame = {
+          ...state.myOpponentGame,
+          id: null,
+        };
+      }
+
       return {
         ...state,
         isGameEnd: action.payload,
+        ...setGameEndUpdate,
+      };
+    case AT.CLEAR_FIELDS:
+      return {
+        ...state,
```

Comment:
...state.initialState remove all other

Reply:
@alexeysanin  we don't need to reset all the state here, only those fields